modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaService.java

```diff
@@ -210,7 +225,14 @@ private <R> CompletableFuture<R> sendToReplica(String targetNodeConsistentId, Re
                             return null;
                         });
                     } else {
-                        res.completeExceptionally(errResp.throwable());
+                        if (retryExecutor != null && matchAny(unwrapCause(errResp.throwable()), ACQUIRE_LOCK_ERR, REPLICA_MISS_ERR)) {
```

Comment:
I still think that using exception classes is better than error codes, in general.
By the way, should `ACQUIRE_LOCK_TIMEOUT_ERR` be taken into account as well?

Reply:
>I still think that using exception classes is better than error codes, in general.
I disagree. The whole exception design is based on error codes. Checking error codes is more clean then compartion exception classes.
> By the way, should ACQUIRE_LOCK_TIMEOUT_ERR be taken into account as well?
No it should not. This error code and related functionality should be removed, because we got retries from client side.

Reply:
> I disagree. The whole exception design is based on error codes. Checking error codes is more clean then comparing exception classes.

I disagree, using Java classes that represent an exception is a widespread practice. Error codes are a way to provide the user with an additional clue on critical situations, especially in the case of thin clients that are not supported exceptions.

Reply:
> I disagree, using Java classes that represent an exception is a widespread practice. Error codes are a way to provide the user with an additional clue on critical situations, especially in the case of thin clients that are not supported exceptions.

Looks like we have not agreement here, because changing error code checking to `instanceof ` doesn't make much sense to me. This however have should not block PR changes. Or is this a merge blocker ?

Reply:
This is definitely not a blocker.